packages/react-instantsearch-core/src/connectors/connectVoiceSearch.js

```diff
@@ -0,0 +1,159 @@
+import PropTypes from 'prop-types';
+import createConnector from '../core/createConnector';
+import {
+  cleanUpValue,
+  refineValue,
+  getCurrentRefinementValue,
+  getIndexId,
+} from '../core/indexUtils';
+
+function getId() {
+  return 'query';
+}
+
+function getAdditionalId() {
+  return 'additionalVoiceParameters';
+}
+
+function getCurrentRefinementQuery(props, searchState, context) {
+  const id = getId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return '';
+}
+
+function getCurrentRefinementAdditional(props, searchState, context) {
+  const id = getAdditionalId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return {};
```

Comment:
Not sure about the expected return value here.

In `getCurrentRefinementQuery` the return value can be `currentRefinement` or a string `''`.
In `getCurrentRefinementAdditional` the return value can be `currentRefinement` or an object `{}`.

I don't know what type `currentRefinement` is returning but it seems to me that the return type is at least inconsistent is one of the 2 functions.

Reply:
currentRefinement is a string for the query, and an object for the additional, so it's consistent, no?

Reply:
Ok, as seen together `getCurrentRefinementValue` always return a string when id is `getId()` return value, and always an object with id is `getAdditionalId()` return value. So no inconsistency here :+1: